src/robotFactory.js

```diff
@@ -1,10 +1,109 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
 
-class FlyingRobot {}
+    this.coords = {
+      x: coords.x || 0,
+      y: coords.y || 0,
+    };
+    this.chipVersion = chipVersion;
+  }
 
-class DeliveryDrone {}
+  goForward(value = 1) {
+    if (value <= 0) {
+      return;
+    }
+
+    this.coords.y += value;
+  }
+
+  goBack(value = 1) {
+    if (value <= 0) {
+      return;
+    }
+
+    this.coords.y -= value;
+  }
+
+  goRight(value = 1) {
+    if (value <= 0) {
+      return;
+    }
+
+    this.coords.x += value;
+  }
+
+  goLeft(value = 1) {
+    if (value <= 0) {
+      return;
+    }
+
+    this.coords.x -= value;
+  }
+
+  getInfo() {
+    // eslint-disable-next-line max-len
+    return `Robot: ${this.name}, Chip version: ${this.chipVersion}, Weight: ${this.weight}`;
+  }
+}
+
+class FlyingRobot extends BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    super(name, weight, coords, chipVersion);
+
+    this.coords = {
+      x: coords.x || 0,
+      y: coords.y || 0,
+      z: coords.z || 0,
+    };
```

Comment:
Your base robot already has x and y coords so you don't need to set them again
```suggestion
    this.coords.z: coords.z || 0;
```